### PR TITLE
Move the AllSitesIcon component under blocks/all-sites, migrate CSS

### PR DESCRIPTION
**Pull request by jsnajdr**
Moves the `AllSitesIcon` component under `blocks/all-sites`. It was a bit misplaced in `my-sites` and it's not really used as an independent component.

Migrates CSS to webpack and does a few little refactors like using arrow functions inside `filter` and `map`.

**How to test:**
Verify the styling of the component in the site selector:

screenshot 2019-01-29 at 15 39 07

### CODE REVIEW

File: client/my-sites/all-sites-icon/README.md [hunk 1]
```diff
@@ -1,22 +0,0 @@
-All Sites Icon
```
Comment (blowery) [inline client/my-sites/all-sites-icon/README.md]: why'd this get dropped?

**Review by blowery — APPROVED** [review]
Code and style moves look good, all sites renders as expected. 

Just curious why the README got dropped. Feel free to land either way.

File: client/my-sites/all-sites-icon/README.md (hunk 1 above)
Comment (jsnajdr) [reply client/my-sites/all-sites-icon/README.md]: Where would I put the README? `AllSitesIcon` is now an internal component of `AllSites`. It was never used independently. I'm not sure if it was ever supposed to.

There is still a README for the `block/all-sites` block. Do you think the deleted README contains some valuable info that should be copied to the parent component README? Or to a comment in code?

Comment (blowery) [reply client/my-sites/all-sites-icon/README.md]: ahhhhh i see. nevermind!

**Pull request merged**